src/warnet/k8s.py

```diff
@@ -115,3 +115,21 @@ def get_default_namespace() -> str:
     command = "kubectl config view --minify -o jsonpath='{..namespace}'"
     kubectl_namespace = run_command(command)
     return kubectl_namespace if kubectl_namespace else DEFAULT_NAMESPACE
+
+
+def wait_for_caddy_ready(name, namespace, timeout=300):
```

Comment:
this also feels a bit smelly to me, as this is what helm is supposed to be managing for us. I didn't see where this is being used, but im guessing it's for the port forwarding?

Reply:
Yeah, sorry if my comment was unclear, more saying this `wait_for_caddy` function seems _specific_ to port forwarding, so we wouldn't need it if we weren't using port forwarding.